Bottom: 56cbb0abc9747fb2ca103201fff1eab727ea41fe
Top:    d40415b72ef6d25d6bc8bc7b424cbd82b8df8e58
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2013-01-03 02:37:57 +0000

configure.ac: Save and restore CPPFLAGS either side of test for X11 stuff

We add X_CFLAGS and X_LIBS to the main flag list later, so don't leave
an extra copy lying around in the CPPFLAGS variable.


---

diff --git a/configure.ac b/configure.ac
index 5fe1a81..fd16e85 100644
--- a/configure.ac
+++ b/configure.ac
@@ -704,6 +704,7 @@ for e in $HIDLIST; do
     case $e in
       lesstif )
 	AC_PATH_XTRA
+	save_CPPFLAGS="$CPPFLAGS"
 	CPPFLAGS="$X_CFLAGS"
 	AC_CHECK_LIB(X11, XOpenDisplay, , , $X_LIBS)
 	AC_CHECK_LIB(ICE, main, , , $X_LIBS)
@@ -712,7 +713,8 @@ for e in $HIDLIST; do
 	AC_CHECK_LIB(Xt, XtOpenDisplay, , , $X_LIBS)
 	AC_CHECK_LIB(Xmu, main, , , $X_LIBS)
 	AC_CHECK_LIB(Xpm, main, , , $X_LIBS)
-        AC_CHECK_LIB(Xm, XmCreateMainWindow, , , $X_LIBS)
+	AC_CHECK_LIB(Xm, XmCreateMainWindow, , , $X_LIBS)
+	CPPFLAGS="$save_CPPFLAGS"
 	case $ac_cv_lib_Xm_XmCreateMainWindow in
 	  no )
 	    AC_MSG_ERROR([You don't seem to have the Lesstif development environment installed.])
